x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -173,69 +174,169 @@ private AuthorizationInfo getAuthorizationInfoFromContext() {
-                    auditId = AuditUtil.getOrGenerateRequestId(threadContext);
-                } else {
-                    auditTrailService.get().tamperedRequest(null, authentication, action, originalRequest);
-                    final String message = "Attempt to authorize action [" + action + "] for [" + authentication.getUser().principal()
-                            + "] without an existing request-id";
-                    assert false : message;
-                    listener.onFailure(new ElasticsearchSecurityException(message));
-                    return;
-                }
-            }
-
-            // sometimes a request might be wrapped within another, which is the case for proxied
-            // requests and concrete shard requests
-            final TransportRequest unwrappedRequest = maybeUnwrapRequest(authentication, originalRequest, action, auditId);
-
-            // Check operator privileges
-            // TODO: audit?
-            final ElasticsearchSecurityException operatorException =
-                operatorPrivilegesService.check(action, originalRequest, threadContext);
-            if (operatorException != null) {
-                listener.onFailure(denialException(authentication, action, originalRequest, operatorException));
-                return;
-            }
-            operatorPrivilegesService.maybeInterceptRequest(threadContext, originalRequest);
-
             if (SystemUser.is(authentication.getUser())) {
                 // this never goes async so no need to wrap the listener
                 authorizeSystemUser(authentication, action, auditId, unwrappedRequest, listener);
             } else {
-                final String finalAuditId = auditId;
-                final RequestInfo requestInfo = new RequestInfo(authentication, unwrappedRequest, action);
                 final ActionListener<AuthorizationInfo> authzInfoListener = wrapPreservingContext(ActionListener.wrap(
                         authorizationInfo -> {
                             threadContext.putTransient(AUTHORIZATION_INFO_KEY, authorizationInfo);
-                            maybeAuthorizeRunAs(requestInfo, finalAuditId, authorizationInfo, listener);
+                            maybeAuthorizeRunAs(requestInfo, auditId, authorizationInfo, listener);
                         }, listener::onFailure), threadContext);
-                getAuthorizationEngine(authentication).resolveAuthorizationInfo(requestInfo, authzInfoListener);
+                engine.resolveAuthorizationInfo(requestInfo, authzInfoListener);
+            }
+        }
+    }
+
+    /**
+     * @return {@code true} if this action is a child action of an already authorized action that is in the thread context
+     * <strong>and</strong> we can infer that this action is authorized from the parent action.
+     */
+    private boolean shouldAuthorizeAsChildAction(AuthorizationEngine engine, RequestInfo requestInfo) {
```

Comment:
I don't think it can happen in today's code. So this is more of a philosophical question: Do we need to worry about the situation where a child action has a different authentication than the parent, but somehow the same authorization info. A possible way to get to this situation is mis-use of threadContext or other helper classes, e.g. ClientHelper and does not clear all associated context entirely. To prevent this, maybe we can add an assert in `AuthenticationService#writeAuthToContext` to ensure when a new authentication is serialised, there should be no authorization info>

Reply:
Would it matter? If it's the same AuthzInfo then the outcome should be the same...